### PR TITLE
fix issue #405 empty csv error

### LINKED ISSUE #405: Get NullPointerException when iterate from empty csv file
When I iterate with source is an empty car.csv file. I got NullPointerException.
```
    <iterate type="problems" count="10" source="car.csv" consumer="ConsoleExporter">
        <attribute name="test"/>
    </iterate>
```

Error look like this:
```
Caused by: java.lang.NullPointerException
	at com.rapiddweller.platform.csv.CSVEntityIterator.init(CSVEntityIterator.java:178)
	at com.rapiddweller.platform.csv.CSVEntityIterator.assureInitialized(CSVEntityIterator.java:165)
	at com.rapiddweller.platform.csv.CSVEntityIterator.next(CSVEntityIterator.java:143)
	at com.rapiddweller.benerator.wrapper.DataSourceGenerator.generate(DataSourceGenerator.java:109)
	at com.rapiddweller.benerator.wrapper.GeneratorWrapper.generateFromSource(GeneratorWrapper.java:70)
	at com.rapiddweller.benerator.wrapper.ConvertingGenerator.generate(ConvertingGenerator.java:88)
	at com.rapiddweller.benerator.wrapper.GeneratorProxy.generate(GeneratorProxy.java:74)
	at com.rapiddweller.benerator_ee.SynchronizedGenerator.generate(SynchronizedGenerator.java:67)
	at com.rapiddweller.benerator.engine.CurrentProductGeneration.execute(CurrentProductGeneration.java:59)
	at com.rapiddweller.benerator.engine.statement.GenIterTask.execute(GenIterTask.java:169)
	... 46 more
```

### PR DESCRIPTION
**Pull request by tunglxfast**
fix issue #405 empty .csv file error, also fix some tests

### CODE REVIEW

File: src/demo/resources/demo/db/postgres.types.ben.xml [hunk 1]
```diff
@@ -25,8 +25,7 @@
         "text"      TEXT NOT NULL,
         "binary"    BYTEA NOT NULL,
         "json"      JSON NOT NULL,
-        "jsonb"     JSONB NOT NULL,
-        "genom"     GEOMETRY NOT NULL
```
Comment (ake2l) [inline src/demo/resources/demo/db/postgres.types.ben.xml]: ??? why did you remove genom

File: src/demo/resources/demo/shop/shop-postgres-mongodb.ben.xml [hunk 2]
```diff
@@ -59,7 +59,6 @@
     <comment>define a mongodb database that will be referred by the id 'mongodb' subsequently</comment>
     <mongodb id="mongodb" host="{ftl:${mongoHost}}" port="{ftl:${mongoPort}}" database="{ftl:${mongoDatabase}}"
              user="{ftl:${mongoUser}}" password="{ftl:${mongoPassword}}"
-             authenticationDatabase="{ftl:${mongoAuthenticationDatabase}}"/>
```
Comment (ake2l) [inline src/demo/resources/demo/shop/shop-postgres-mongodb.ben.xml:62]: here the same , doesn't seem to have to do anything with the csv error

File: src/test/java/com/rapiddweller/benerator/engine/DemoIntegrationNoExtDBTest.java [hunk 3]
```diff
@@ -311,4 +311,14 @@ public void watermark() {
     Assert.assertEquals("/demo/watermark", benCtx.getContextUri());
   }
 
+  /**
+   * Test files empty csv.
+   */
+  @Test
+  public void testFilesEmptyCSV() {
```
Comment (ake2l) [inline src/test/java/com/rapiddweller/benerator/engine/DemoIntegrationNoExtDBTest.java:318]: great that you added a test as well

File: src/demo/resources/demo/db/postgres.types.ben.xml (hunk 1 above)
Comment (tunglxfast) [reply src/demo/resources/demo/db/postgres.types.ben.xml]: I remove it for my local test but not forgot to add again. I will change it back. I test it because sometime GEOMETRY type will cause ERROR:  type "geometry" does not exist. That happen when the user postgers don't have postgis EXTENSION (can be fix by code: CREATE EXTENSION postgis;)

File: src/demo/resources/demo/shop/shop-postgres-mongodb.ben.xml (hunk 2 above)
Comment (tunglxfast) [reply src/demo/resources/demo/shop/shop-postgres-mongodb.ben.xml:62]: Yes, it doesn't have relative to the csv error. I just see the code wrong went having 2 close tag /> so it change it.

Comment (ake2l) [reply src/demo/resources/demo/shop/shop-postgres-mongodb.ben.xml:62]: good point alright

File: src/demo/resources/demo/db/postgres.types.ben.xml (hunk 1 above)
Comment (ake2l) [reply src/demo/resources/demo/db/postgres.types.ben.xml]: yes we need to fix it ...

`postgis EXTENSION (can be fix by code: CREATE EXTENSION postgis;)`
